src/stpipe/config.py

```diff
@@ -111,7 +111,10 @@ def to_asdf(self, include_metadata=False):
             meta = deepcopy(_META_TEMPLATE)
             meta["date"] = meta["date"].replace(
                 _TEMPLATE_PLACEHOLDER,
-                datetime.utcnow().replace(microsecond=0).isoformat(),
+                datetime.now(UTC)
```

Comment:
```suggestion
                datetime.now(timezone.utc)
```
I believe this and the other suggestion should allow the CI to work with 3.9 and matches the [changes made in jwst](https://github.com/spacetelescope/jwst/pull/8051/files#diff-d0a38e200cb582a0443f0e448a2eb55c044ab229869df889ed7e5c633facb8b5R567).

Although not necessary with this PR, would you add a python 3.12 environment to tox and the CI? This would be great to verify that the `DeprecationWarning` is fixed.

Reply:
Thanks.  Done.

This actually doesn't get picked up in the `stpipe` unit tests, but does in the `jwst` unit tests, which test `config_parser.py` better than `stpipe` does.  One might want to think about adding coverage to the downstream tests, specifically `jwst`.